### ISSUE TITLE
We have Violinplots but no kdeplot.   

### ISSUE THREAD

**Issue by jklymak** [matplotlib/matplotlib#17341]
Someone  on twitter pointed out that its pretty strange that we have a KDE estimator that we use for `violinplot`, (`mlab.GaussianKDE`) but then we tell the user to draw their own pdf if they just want to plot the thing as an x-y plot.  

I think `violinplot` probably should have been left to downstream packages like seaborn.  I think its confusing to users to tell them that we support this sort of smoothing of distributions for one type of statistical plot, but not simpler ones.

**Comment by story645** [comment]
Looking at the violinplots PR (#2996) , it's apparently @tacaswell fault - back then he apparently wanted matplotlib to carry basic stats plots. 

I'm with you - really we should deprecate violinplot but since we're not we should add kde plot. Maybe implement that one under a new provisional stats namespace?

**Comment by jklymak** [comment]
Yeah, thats a bit of a big job and executive decision if we are moving to name spaces.  If we have a new namespace, we should probably should populate it with a few other things - `boxplot` also comes to mind.  Of course we'd still need to echo those in `_axes.py`.

**Comment by tacaswell** [comment]
This also seems related to the recent ECDF (https://github.com/matplotlib/matplotlib/issues/16561) discussion.

Do we understand the right way to paramatarize a kdeplot?  Are we going to get ourselves into a prolonged discussion about how to do statistics?  If we add `kdeplot` what is going to be "on deck" as the rest of the camel (how do you like those mixed metaphors?)?

**Comment by WeatherGod** [comment]
Couldn't we just see violin and kde plots as two sides of the coin and move
what would be common parts into a private module?

**Comment by story645** [comment]
I think if we go this route, I think the stats plots should be coded against a common reference (my vote is for [NIST](https://www.itl.nist.gov/div898/software/dataplot/refman1/auxillar/kernplot.htm)) and probably moved to a provisional namespace.
